builtin/credential/okta/path_config.go

```diff
@@ -0,0 +1,148 @@
+package okta
+
+import (
+	"fmt"
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	"github.com/sstarcher/go-okta"
+	"net/url"
+)
+
+func pathConfig(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: `config`,
+		Fields: map[string]*framework.FieldSchema{
+			"organization": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Okta organization to authenticate against",
+			},
+			"token": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Okta admin API token",
+			},
+			"base_url": &framework.FieldSchema{
+				Type: framework.TypeString,
+				Description: `The API endpoint to use. Useful if you
+are using Okta development accounts.`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathConfigRead,
+			logical.UpdateOperation: b.pathConfigWrite,
+		},
+
+		HelpSynopsis: pathConfigHelp,
+	}
+}
+
+// Config returns the configuration for this backend.
+func (b *backend) Config(s logical.Storage) (*ConfigEntry, error) {
+	entry, err := s.Get("config")
+	if err != nil {
+		return nil, err
+	}
+
+	var result ConfigEntry
+	if entry != nil {
+		if err := entry.DecodeJSON(&result); err != nil {
+			return nil, err
+		}
+	}
+
+	return &result, nil
+}
+
+func (b *backend) pathConfigRead(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	cfg, err := b.Config(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if cfg == nil {
+		return nil, nil
+	}
+
+	resp := &logical.Response{
+		Data: structs.New(cfg).Map(),
+	}
+
+	return resp, nil
+}
+
+func (b *backend) pathConfigWrite(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	org := d.Get("organization").(string)
+	token := d.Get("token").(string)
+	baseURL := d.Get("base_url").(string)
+	if len(baseURL) != 0 {
+		_, err := url.Parse(baseURL)
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf("Error parsing given base_url: %s", err)), nil
+		}
+	}
+
+	entry, err := logical.StorageEntryJSON("config", ConfigEntry{
+		Org:     org,
+		Token:   token,
+		BaseURL: baseURL,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if err := req.Storage.Put(entry); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (c *ConfigEntry) OktaClient() *okta.Client {
+	client := okta.NewClient(c.Org)
+	if c.BaseURL != "" {
+		client.Url = c.BaseURL
+	}
+
+	if c.Token != "" {
+		client.ApiToken = c.Token
+	}
+
+	return client
+}
+
+type ConfigEntry struct {
+	Org     string `json:"organization"`
+	Token   string `json:"token"`
+	BaseURL string `json:"base_url"`
+}
+
+/*
+ * Returns FieldData describing our ConfigEntry struct schema
+ */
+func (b *backend) getConfigFieldData() (*framework.FieldData, error) {
```

Comment:
This doesn't appear to be used anywhere. I'm guessing this was copied over from LDAP?